### ISSUE TITLE
Recovery flow breaks with emails containing uppercase characters

### ISSUE THREAD

**Issue by nkeenan38** [ory/kratos#2426]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/kratos/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/kratos/blob/master/CONTRIBUTING.md).
- [ ] This issue affects my [Ory Cloud](https://www.ory.sh/) project.
- [X] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [X] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe the bug

When an identity is created with an email containing uppercase characters, most self-service flow correctly behave and ignore casing. It seems that uppercase letter break the recovery flow. No matter the casing of the input, the recovery flow always sends the invalid recovery email template. This happens even when the value in `identity_recovery_addresses` has the same casing as the trait in the email.

### Reproducing the bug

1. Create a new user with an email containing uppercase characters through self-service registration.
2. Verify the email if necessary in the configuration.
3. Try the self service recovery flow with the email used.

The email in question should receive the invalid recovery email template.

### Version

v0.9.0

### On which operating system are you observing this issue?

Linux

### In which environment are you deploying?

Kubernetes

**Comment by aeneasr** [comment]
Thank you for the report - I believe that this was fixed but it is possible that it wasn't. Are you 100% certain to be on the latest version of Ory Kratos? You can confirm this using `kratos version`. Thanks!

**Comment by kszafran** [comment]
I have just encountered a similar problem. It worked in 0.8.2. I suspect the problem is that the code has been changed in 0.9.0 to store recovery and verifiable email addresses in lowercase, but migrations haven't converted existing addresses. So users created with uppercase chars before upgrading to 0.9.0 can't go through the recovery flow. It can be fixed by updating addresses in the DB:
```sql
update identity_recovery_addresses set value=lower(value);
update identity_verifiable_addresses set value=lower(value);
```
